lib/SILGen/SILGenBridging.cpp

```diff
@@ -718,6 +718,12 @@ static ManagedValue emitNativeToCBridgedNonoptionalValue(SILGenFunction &SGF,
 
   // Fall back to dynamic Any-to-id bridging.
   // The destination type should be AnyObject in this case.
+  if (!bridgedType->isEqual(SGF.getASTContext().getAnyObjectType())) {
+    llvm::errs() << "Fallback not viable. About to crash. \n";
```

Comment:
would it be better to have a user-visible frontend error that prohibits this bridging?